packages/blade-svelte/src/components/Link/BaseLink/BaseLink.svelte

```diff
@@ -0,0 +1,304 @@
+    const state = currentInteraction;
+    const map: Record<ActionStatesType, ColorType> = {
+      default: 'normal',
+      hover: 'subtle',
+      focus: 'normal',
+      disabled: 'disabled',
+    };
+
+    const stateKey = map[state];
+    
+    if (color && color !== 'primary') {
+      if (color !== 'white') {
+        return `var(--interactive-${element}-${color}-${stateKey})`;
+      }
+      return `var(--interactive-${element}-static-white-${stateKey})`;
+    }
+    return `var(--interactive-${element}-primary-${stateKey})`;
+  }
+
+  // Get text size tokens
+  const textSizes = {
+    fontSize: {
+      xsmall: 'var(--font-size-25)',
+      small: 'var(--font-size-75)',
+      medium: 'var(--font-size-100)',
+      large: 'var(--font-size-200)',
+    },
+    lineHeight: {
+      xsmall: 'var(--line-height-25)',
+      small: 'var(--line-height-75)',
+      medium: 'var(--line-height-100)',
+      large: 'var(--line-height-200)',
+    },
+  };
+
+  // Reactive values based on currentInteraction - must use $derived to be reactive
+  const textDecorationLine = $derived(!isButton && currentInteraction !== 'default' ? 'underline' : 'none');
+  const iconColor = $derived(getColorToken('icon'));
+  const textColor = $derived(getColorToken('text'));
+  const iconPadding = children ? 'var(--spacing-2)' : 'var(--spacing-0)';
+  const defaultRel = target && target === '_blank' ? 'noreferrer noopener' : undefined;
+  
+  // Cursor logic: always 'pointer' unless it's a button variant AND disabled (then 'not-allowed')
+  const cursor = $derived(disabled ? 'not-allowed' : 'pointer');
+
+  // Accessibility attributes
+  const accessibilityAttrs = makeAccessible({
+    role: isButton ? 'button' : 'link',
+    disabled: isButton && isDisabled,
+    ...accessibilityProps,
+  });
+
+  // Meta attributes
+  const metaAttrs = metaAttribute({
+    name: MetaConstants.Link,
+    testID,
+  });
+
+  // Analytics attributes
+  const analyticsAttrs = makeAnalyticsAttribute(rest);
+
+  // Event handlers
+  function handleClick(event: MouseEvent): void {
+    if (onClick && !disabled) {
+      onClick(event);
+    }
+  }
+
+  function handleFocus(event: FocusEvent): void {
+    if (!disabled) {
+      handleFocusInteraction();
+    }
+    onFocus?.(event);
+  }
+
+  function handleBlur(event: FocusEvent): void {
+    if (!disabled) {
+      handleBlurInteraction();
+    }
+    onBlur?.(event);
+  }
+
+  function handleMouseEnter(): void {
+    if (!disabled) {
+      handleMouseEnterInteraction();
+    }
+  }
+
+  function handleMouseLeave(event: MouseEvent): void {
+    if (!disabled) {
+      handleMouseLeaveInteraction();
+    }
+    onMouseLeave?.(event);
+  }
+
+  // Computed styles
+  const linkClass = `base-link base-link--${variant} base-link--${color} base-link--${size}`;
+  const contentClass = 'base-link__content';
+  const textClass = `base-link__text base-link__text--${size}`;
+</script>
+
+{#if isButton}
+  <button
```

Comment:
is there any equivalent of `as` prop in svelte that we can use or implement in these cases?

`<BaseBox as={isButton ? 'button' : 'a'}>`